src/scales/scale.time.js

```diff
@@ -403,6 +403,27 @@ function ticksFromTimestamps(values, majorUnit) {
 	return ticks;
 }
 
+function determineLabelFormat(data, timeOpts) {
+	var i, momentDate, hasTime;
+	var ilen = data.length;
+
+	// find the label with the most parts (milliseconds, minutes, etc.)
+	// format all labels with the same level of detail as the most specific label
+	for (i = 0; i < ilen; i++) {
+		momentDate = momentify(data[i], timeOpts);
+		if (momentDate.millisecond() !== 0) {
+			return 'MMM D, YYYY h:mm:ss.SSS a';
+		}
+		if (momentDate.second() !== 0 || momentDate.minute() !== 0 || momentDate.hour() !== 0) {
```

Comment:
what happens at `00:00:00` ie, midnight on a day? Or is there no way to detect a time set with that from a `moment` object that has no time at all?

Reply:
The chart.js user is specifying timestamps if we're utilizing this function. With timestamps there's no way to differentiate dates vs date plus time of midnight. However, if every single timestamp in the dataset is midnight then it's relatively clear that they're representing dates. Also, I don't know that there's any practical difference between date vs date+midnight.

Reply:
True, if there is at least one time other than midnight, then it will pick `'MMM D, YYYY h:mm:ss a'`, else if all times are midnight then it will pick `'MMM D, YYYY'`, which is ok I guess.